### ISSUE TITLE
use boolean instead of int to indicate where golint is applied and to know whether there is one arg

### ISSUE THREAD

**Issue by tomocy** [golang/lint#432]
At golint.go:48
```go
var dirsRun, filesRun, pkgsRun int
```
These parameters are used to indicate where golint is applied and to know whether there are args.
```go
if dirsRun+filesRun+pkgsRun != 1 {
    // ...
}
switch {
case dirsRun == 1:
    // ...
case filesRun == 1:
    // ...
case pkgsRun == 1:
    // ...
}
```

I think it can be replaced by boolean and that this change will make code more readable.
- to indicate where golint is applied
```go
switch {
case dirsRun:
    // ...
case filesRun:
    // ...
case pkgsRun:
    // ...
}
```
- to know whether there is one arg or none
```go
if needUsage(dirsRun, filesRun, pkgsRun) {
}

func needUsage(runs ...bool) bool {
   // return true if runs are mutually exclusively true
}
```

**Comment by mvdan** [comment]
Thank you for submitting this issue! As per https://github.com/golang/go/issues/38968, we are freezing and deprecating golint. There's no drop-in replacement to golint per se, but you should find that [Staticcheck](https://staticcheck.io/) works well in encouraging good Go code, much like golint did in the past, since it also includes style checks. There's always `gofmt` and `go vet` too, of course.

If you would like to contribute further, I'd encourage you to engage [Staticcheck's issue tracker](https://github.com/dominikh/go-tools/issues) or look at [vet's open issues](https://github.com/golang/go/issues?q=is%3Aissue+is%3Aopen+cmd%2Fvet+in%3Atitle), as they are both actively maintained. If you have an idea that doesn't fit into either of those tools, you could look at [other Go linters](https://github.com/golangci/awesome-go-linters), or write your own - these days it's fairly straightforward with [go/analysis](https://pkg.go.dev/golang.org/x/tools/go/analysis).

To help avoid confusion, I'm closing all issues before we freeze the repository. If you have any feedback, you can leave a comment on the proposal thread where it was decided to deprecate golint - though note that the proposal has been accepted for nearly a year. Thanks!
